plugins/inputs/mcrouter/mcrouter.go

```diff
@@ -0,0 +1,224 @@
+  servers = ["localhost:11211"]
+  # unix_sockets = ["/var/run/mcrouter.sock"]
+`
+
+var defaultTimeout = 5 * time.Second
+
+// The list of metrics that should be sent
+var sendMetrics = []string{
+	"uptime",
+	"num_servers",
+	"num_servers_new",
+	"num_servers_up",
+	"num_servers_down",
+	"num_servers_closed",
+	"num_clients",
+	"num_suspect_servers",
+	"destination_batches_sum",
+	"destination_requests_sum",
+	"outstanding_route_get_reqs_queued",
+	"outstanding_route_update_reqs_queued",
+	"outstanding_route_get_avg_queue_size",
+	"outstanding_route_update_avg_queue_size",
+	"outstanding_route_get_avg_wait_time_sec",
+	"outstanding_route_update_avg_wait_time_sec",
+	"retrans_closed_connections",
+	"destination_pending_reqs",
+	"destination_inflight_reqs",
+	"destination_batch_size",
+	"asynclog_requests",
+	"proxy_reqs_processing",
+	"proxy_reqs_waiting",
+	"client_queue_notify_period",
+	"rusage_system",
+	"rusage_user",
+	"ps_num_minor_faults",
+	"ps_num_major_faults",
+	"ps_user_time_sec",
+	"ps_system_time_sec",
+	"ps_vsize",
+	"ps_rss",
+	"fibers_allocated",
+	"fibers_pool_size",
+	"fibers_stack_high_watermark",
+	"successful_client_connections",
+	"duration_us",
+	"destination_max_pending_reqs",
+	"destination_max_inflight_reqs",
+	"retrans_per_kbyte_max",
+	"cmd_get_count",
+	"cmd_delete_out",
+	"cmd_lease_get",
+	"cmd_set",
+	"cmd_get_out_all",
+	"cmd_get_out",
+	"cmd_lease_set_count",
+	"cmd_other_out_all",
+	"cmd_lease_get_out",
+	"cmd_set_count",
+	"cmd_lease_set_out",
+	"cmd_delete_count",
+	"cmd_other",
+	"cmd_delete",
+	"cmd_get",
+	"cmd_lease_set",
+	"cmd_set_out",
+	"cmd_lease_get_count",
+	"cmd_other_out",
+	"cmd_lease_get_out_all",
+	"cmd_set_out_all",
+	"cmd_other_count",
+	"cmd_delete_out_all",
+	"cmd_lease_set_out_all",
+}
+
+// SampleConfig returns sample configuration message
+func (m *Mcrouter) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns description of Mcrouter plugin
+func (m *Mcrouter) Description() string {
+	return "Read metrics from one or many mcrouter servers"
+}
+
+// Gather reads stats from all configured servers accumulates stats
+func (m *Mcrouter) Gather(acc telegraf.Accumulator) error {
+	if len(m.Servers) == 0 && len(m.UnixSockets) == 0 {
+		return m.gatherServer(":11211", false, acc)
+	}
+
+	for _, serverAddress := range m.Servers {
+		acc.AddError(m.gatherServer(serverAddress, false, acc))
+	}
+
+	for _, unixAddress := range m.UnixSockets {
+		acc.AddError(m.gatherServer(unixAddress, true, acc))
+	}
+
+	return nil
+}
+
+func (m *Mcrouter) gatherServer(
+	address string,
+	unix bool,
+	acc telegraf.Accumulator,
+) error {
+	var conn net.Conn
+	var err error
+	if unix {
+		conn, err = net.DialTimeout("unix", address, defaultTimeout)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+	} else {
+		_, _, err = net.SplitHostPort(address)
+		if err != nil {
+			address = address + ":11211"
+		}
+
+		conn, err = net.DialTimeout("tcp", address, defaultTimeout)
```

Comment:
Would be a nice touch to create a context.Context at the beginning of Gather, and use DialContext, I recently updated the zookeeper input to do something like this:

https://github.com/influxdata/telegraf/blob/master/plugins/inputs/zookeeper/zookeeper.go#L81

Reply:
I think I followed the zookeeper example here